File: Oxygene/Cooper/Android/HelloWorldManualUI/MainActivity.pas
namespace org.me.helloworld.manualui;

//Sample app by Brian Long (http://blong.com)

{
  This example is a simple Android application with a button that displays a dialog
  However there is no UI layout file - the (admittedly simple) UI is built up in code
}

interface

uses
  java.util,
  java.text,
  android.os,
  android.app,
  android.util,
  android.view,
  android.content,
  android.widget;
  
type
  MainActivity = public class(Activity)
  private
    //Identify the dialog
    const HelloDialog = 1;
    var helloButton: Button;   
  protected
    //Called the first time the dialog is displayed
    method onCreateDialog(Id: Integer): Dialog; override;
    //Called every time the dialog is displayed
    method onPrepareDialog(Id: Integer; Dlg: Dialog); override;
  public
    method onCreate(savedInstanceState: Bundle); override;
  end;

implementation

method MainActivity.onCreate(savedInstanceState: Bundle);
begin
  inherited;
  var mainLayout := new LinearLayout(Self);
  //Make a linear layout that fills the screen and centres child views horizontally & vertically
  mainLayout.Orientation := LinearLayout.VERTICAL;
  mainLayout.LayoutParams := new LinearLayout.LayoutParams(
      LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT);
  mainLayout.Gravity := Gravity.CENTER_VERTICAL or Gravity.CENTER_HORIZONTAL;

  //Make a button as tall.wide as its content dictates
  helloButton := new Button(Self);
  helloButton.Text := String[R.string.hello_button_text];
  helloButton.LayoutParams := new ViewGroup.LayoutParams(
      ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);

  //Add button to the layout and make that our UI
  mainLayout.addView(helloButton);
  ContentView := mainLayout;
  helloButton.OnClickListener := -> showDialog(HelloDialog);
end;

//This is called the first time the dialog is displayed
method MainActivity.onCreateDialog(Id: Integer): Dialog;
begin
  Result := inherited;
  if Id = HelloDialog then
  begin
    var builder := new AlertDialog.Builder(Self);
    //Put a default message on the dialog (though this will be overwritten in onPrepareDialog)
    builder.Message := R.string.hello_dialog_message;
    //Don't allow Back to get rid of the dialog
    builder.Cancelable := False;
    //Set up the positive response button, which happens to be the only button
    builder.setPositiveButton(R.string.hello_dialog_button_text, (s, a) -> (s as Dialog).dismiss);
    //Make the dialog
    Result := builder.&create;
    //Do something when the dialog is dismissed - update the activity's button
    Result.OnDismissListener := (d) -> begin helloButton.Text := String[R.string.hello_button_text_2] end;
  end;
end;

//This is called every time the dialog is displayed
method MainActivity.onPrepareDialog(Id: Integer; Dlg: Dialog);
begin
  if Id = HelloDialog then
  begin
    //Update the dialog's message
    var msg := Dlg.findViewById(Android.R.id.message) as TextView;
    if msg <> nil then
      msg.Text := String[R.string.hello_dialog_message_2] + DateFormat.DateTimeInstance.format(new Date);
  end
end;

end.

File: Silver/All/Calculator/Calculator.Android/MainActivity.swift
import java.util
import  android.app
import  android.content
import  android.os
import  android.util
import  android.view
import  android.widget

public class MainActivity: Activity {

	private var TbValue: TextView!
	func btnBackspace(v: View!) {
		var s = TbValue.Text.toString()
		if s.length() > 0 {
			s = s.substring(0, s.length() - 1)
			TbValue.Text = s
		}
	}

	func btnExecute(_ v: View!) {
		__try {
			var eval = Evaluator()
			var res = eval.Evaluate(TbValue.Text.toString())
			TbValue.Text = res.toString()
		} __catch e: Exception {
			var dlg = AlertDialog.Builder(self)
			dlg.Message = "Error evaluating: " + e.Message
			dlg.setPositiveButton("OK", nil)
			dlg.Cancelable = true
			dlg.create().show()
		}
	}

	func btnExit(_ v: View!) {
		finish();
	}

	func btnChar(_ v: View!) {
		TbValue.Text = TbValue.Text.toString() + (v as! Button).Text.toString()
	}

	public override func onCreate(_ savedInstanceState: Bundle!) {
		super.onCreate(savedInstanceState)
		ContentView = R.layout.main
		TbValue = findViewById(R.id.tbValue)as!TextView
		(findViewById(R.id.btnBackspace)as!Button).OnClickListener = btnBackspace
		(findViewById(R.id.btnExit)as!Button).OnClickListener = btnExit
		(findViewById(R.id.btnEval)as!Button).OnClickListener = btnExecute
		(findViewById(R.id.btn0)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btn1)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btn2)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btn3)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btn4)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btn5)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btn6)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btn7)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btn8)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btn9)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btnAdd)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btnSub)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btnDiv)as!Button).OnClickListener = btnChar
		(findViewById(R.id.btnMul)as!Button).OnClickListener = btnChar
	}
}
